aten/src/ATen/native/native_functions.yaml

```diff
@@ -5450,6 +5450,13 @@
     CPU: foreach_add_scalar_kernel_fallback
     CUDA: foreach_tensor_add_scalar_kernel_cuda
 
+- func: _foreach_add_.Scalar(Tensor[](a!) self, Scalar scalar) -> Tensor[](a!)
```

Comment:
we don't have in-place functions, only methods.  Is it feasible to make this an out-variant function?

Reply:
actually this isn't quite true, a bunch of stuff in nn::functional is defined this way (although they aren't typically directly accessible).

Reply:
sounds good then!

Reply:
wouldn't `Tensor[](a!)` mean that the list is being mutated, and `Tensor(a!)[]` be a list of mutable tensors?

@suo is that right?  Is this safe to add?

Reply:
Coming in here late, I noticed this problem too while working on the new codegen at https://github.com/pytorch/pytorch/pull/42629 I also agree that `Tensor(a!)[]` is the correct type, and what was landed (`Tensor[](a!)`) is not correct.

Reply:
I added a small fix for the problem in the new codegen. Besides

```
- - func: _foreach_add_.Scalar(Tensor[](a!) self, Scalar scalar) -> ()
-   device_guard: False  
+ - func: _foreach_add_.Scalar(Tensor(a!)[] self, Scalar scalar) -> ()
+   device_guard: False
```

I also had to add a special case to make void return for trailing underscore ok:

```

        if self.name.name.inplace:
            # TODO: fixme
            if str(self.name) not in [
                    '_amp_non_finite_check_and_unscale_',
                    '_foreach_add_.Scalar']:
                assert len(self.returns) == 1
```